statement summary: update statement summary doc (#6084)
pingcap/docs: statement-summary-tables.md

- [`statements_summary`](#statements_summary)
- [`statements_summary_history`](#statements_summary_history)
- [`cluster_statements_summary`](#statements_summary_evicted)
- [`cluster_statements_summary_history`](#statements_summary_evicted)
- [`statements_summary_evicted`](#statements_summary_evicted)

> - [`statements_summary_evicted`](#statements_summary_evicted)

Author: ```suggestion
```

- [`statements_summary`](#statements_summary)
- [`statements_summary_history`](#statements_summary_history)

> - [`cluster_statements_summary`](#cluster_statements_summary-and-cluster_statements_summary_history)

Author: ```suggestion
- [`cluster_statements_summary`](#statements_summary_evicted)
```

The fields `SUMMARY_BEGIN_TIME` and `SUMMARY_END_TIME` represent the start time and the end time of the historical time range.

## `statements_summary_evicted`

> The `tidb_stmt_summary_max_stmt_count` variable controls the maximum number of statements that the `statement summary` table stores in memory. Using the LRU algorithm, once the types of SQL statements exceed the `tidb_stmt_summary_max_stmt_count` value, the longest missed record is evicted from the table. The types and number of evicted SQL statements during each period are recorded in the `statements_summary_evicted` table.

Author: ```suggestion
The `tidb_stmt_summary_max_stmt_count` variable controls the maximum number of statements that the `statement summary` table stores in memory. The `statement summary` table uses the LRU algorithm. Once the types of SQL statements exceed the `tidb_stmt_summary_max_stmt_count` value, the longest unused record is evicted from the table. The types and number of evicted SQL statements during each period are recorded in the `statements_summary_evicted` table.
```

> The `tidb_stmt_summary_max_stmt_count` variable controls the maximum number of statements that the `statement summary` table stores in memory. The `statement summary` table uses the LRU algorithm. Once the types of SQL statements exceed the `tidb_stmt_summary_max_stmt_count` value, the longest unused record is evicted from the table. The number of evicted SQL statements during each period are recorded in the `statements_summary_evicted` table.

Reviewer: ```suggestion
The `tidb_stmt_summary_max_stmt_count` variable controls the maximum number of statements that the `statement summary` table stores in memory. The `statement summary` table uses the LRU algorithm. Once the number of SQL statements exceed the `tidb_stmt_summary_max_stmt_count` value, the longest unused record is evicted from the table. The number of evicted SQL statements during each period are recorded in the `statements_summary_evicted` table.
```

> The `tidb_stmt_summary_max_stmt_count` variable controls the maximum number of statements that the `statement summary` table stores in memory. The `statement summary` table uses the LRU algorithm. Once the types of SQL statements exceed the `tidb_stmt_summary_max_stmt_count` value, the longest unused record is evicted from the table. The types and number of evicted SQL statements during each period are recorded in the `statements_summary_evicted` table.

Author: ```suggestion
The `tidb_stmt_summary_max_stmt_count` variable controls the maximum number of statements that the `statement summary` table stores in memory. The `statement summary` table uses the LRU algorithm. Once the types of SQL statements exceed the `tidb_stmt_summary_max_stmt_count` value, the longest unused record is evicted from the table. The number of evicted SQL statements during each period are recorded in the `statements_summary_evicted` table.
```

> The `statements_summary_evicted` table is updated only when a SQL record is evicted from the `statement summary` table. The `statements_summary_evicted` only records the period during which the eviction occurs.

Author: ```suggestion
The `statements_summary_evicted` table is updated only when a SQL record is evicted from the `statement summary` table. The `statements_summary_evicted` only records the period during which the eviction occurs and the number of evicted SQL statements.
```

The fields `SUMMARY_BEGIN_TIME` and `SUMMARY_END_TIME` represent the start time and the end time of the historical time range.

## `statements_summary_evicted`

> The `tidb_stmt_summary_max_stmt_count` variable controls the maximum number of statements that the `statement summary` table stores in memory. The `statement summary` table uses the LRU algorithm. Once the number of SQL statements exceed the `tidb_stmt_summary_max_stmt_count` value, the longest unused record is evicted from the table. The number of evicted SQL statements during each period are recorded in the `statements_summary_evicted` table.

Reviewer 2: Should change "`statement summary` table" to "`statements_summary` table"
"`statement summary history` table" -> `statements_summary_history` table

Author: ```suggestion
The `tidb_stmt_summary_max_stmt_count` variable controls the maximum number of statements that the statement summary tables store in memory. The statement summary tables use the LRU algorithm. Once the number of SQL statements exceeds the `tidb_stmt_summary_max_stmt_count` value, the longest unused record is evicted from the tables. The number of evicted SQL statements during each period is recorded in the `statements_summary_evicted` table.
```

> The `statements_summary_evicted` table is updated only when a SQL record is evicted from the `statement summary` table. The `statements_summary_evicted` only records the period during which the eviction occurs and the number of evicted SQL statements.

Author: ```suggestion
The `statements_summary_evicted` table is updated only when a SQL record is evicted from the statement summary tables. The `statements_summary_evicted` only records the period during which the eviction occurs and the number of evicted SQL statements.
```

- `tidb_stmt_summary_refresh_interval`: The interval at which the `statements_summary` table is refreshed. The time unit is second (s). The default value is `1800`.
- `tidb_stmt_summary_history_size`: The size of each SQL statement category stored in the `statements_summary_history` table, which is also the maximum number of records in the `statement_summary_evicted` table. The default value is `24`.
- `tidb_stmt_summary_max_stmt_count`: Limits the number of SQL statements that can be stored in statement summary tables. The default value is `3000`. If the limit is exceeded, those SQL statements that recently remain unused are cleared. These cleared SQL statements are recorded in the `statement_summary_evicted` table.
- `tidb_stmt_summary_max_sql_length`: Specifies the longest display length of `DIGEST_TEXT` and `QUERY_SAMPLE_TEXT`. The default value is `4096`.
- `tidb_stmt_summary_internal_query`: Determines whether to count the TiDB SQL statements. `1` means to count, and `0` means not to count. The default value is `0`.

> **Note:**
>

> > When a category of SQL statement needs to be removed because the `tidb_stmt_summary_max_stmt_count` limit is exceeded, TiDB removes the data of that SQL statement category of all time ranges from the `statement summary history` table. Therefore, even if the number of SQL statement categories in a certain time range does not reach the limit, the number of SQL statements stored in the `statement summary history` table is less than the actual number of SQL statements. If this situation occurs and affects performance, you are recommended to increase the value of `tidb_stmt_summary_max_stmt_count`.

Author: ```suggestion
> When a category of SQL statement needs to be removed because the `tidb_stmt_summary_max_stmt_count` limit is exceeded, TiDB removes the data of that SQL statement category of all time ranges from the `statement_summary_history` table. Therefore, even if the number of SQL statement categories in a certain time range does not reach the limit, the number of SQL statements stored in the `statement_summary_history` table is less than the actual number of SQL statements. If this situation occurs and affects performance, you are recommended to increase the value of `tidb_stmt_summary_max_stmt_count`.
```

>
> The `tidb_stmt_summary_history_size`, `tidb_stmt_summary_max_stmt_count`, and `tidb_stmt_summary_max_sql_length` configuration items affect memory usage. It is recommended that you adjust these configurations based on your needs. It is not recommended to set them too large values.

### Set a proper size for statement summary

> After the system has run for a period of time, you can check the `statement summary` table to see whether SQL eviction has occurred. For example:

Author: ```suggestion
After the system has run for a period of time, you can check the statement summary tables to see whether SQL eviction has occurred. For example:
```

```sql
select @@global.tidb_stmt_summary_max_stmt_count;
select count(*) from information_schema.statements_summary;
```

```sql
+-------------------------------------------+
| @@global.tidb_stmt_summary_max_stmt_count |
+-------------------------------------------+
| 3000                                      |
+-------------------------------------------+
1 row in set (0.001 sec)

+----------+
| count(*) |
+----------+
|     3001 |
+----------+
1 row in set (0.001 sec)
```

> You can see that the `statement summary` table is full of records. Then check the evicted data from the `statements_summary_evicted` table:

Author: ```suggestion
You can see that the statement summary tables are full of records. Then check the evicted data from the `statements_summary_evicted` table:
```
